test/e2e/windows/gmsa_full.go

```diff
@@ -160,7 +160,7 @@ var _ = sigDescribe(feature.Windows, "GMSA Full", framework.WithSerial(), framew
 					return false
 				}
 				return true
-			}, 1*time.Minute, 1*time.Second).Should(gomega.BeTrue())
+			}, 1*time.Minute, 1*time.Second).Should(gomega.BeTrueBecause("Failed to connect to domain for the above error."))
```

Comment:
@pohly I am not sure what message to add in some places because they are already explained in the steps, for e.g. the above one. Please let me know what you think about this so that I can reflect your suggestions in the new and existing changes. Thanks!

Reply:
This is a case where `func() bool` should not be used as callback for `gomega.Eventually`. Instead, return an error and use `gomega.Succeed()`.

All of the log output in the callback can be removed because what's currently getting logged will be in the error and get dumped by gomega if (and only if!) needed.

Reply:
I tried to reflect this change in some more instances. Please let me know if it looks good.